### ISSUE TITLE
tools: Tweak update-repository-version.sh to bump the VERSION of `main` to `alpha0` whenever we branch `rc0``

### ISSUE THREAD

**Issue by fidencio** [kata-containers/kata-containers#1886]
Currently we have an inconsistent behaviour with the `main` branch, as the VERSION file shows:
```
$ cat VERSION 
2.1.0-rc0
```

As `2.1.0` has already been released, this branch should be `2.2.0-alpha0`.

We can manually bump it, of course, but ideally the `update-repository-version.sh` script would do this for us, whenever we switch from `alpha` to `rc`.

**Comment by fidencio** [comment]
@bergwolf, would you have some cycles to take a look at this one?

**Comment by fidencio** [comment]
JFYI, I've done a quick look at this and I'm not sure the best way to proceed.
We need to ensure a few things here.
1. Once we branch `rc0`, which is actually done by the `tag_repos.sh` script, we could run `update-repository-version.sh` passing something like: `next_version=$(echo ${tag} | awk 'BEGIN{FS=OFS="."}{print $1 "." $2+1 ".0-alpha0"}')`, that would kinda work, but ...
2. We need to ensure that once it's done, a simpler commit will be generated, just something like, "post-release bump to ${new_version}", and this should be done in `update-repository-version.sh`.
3. While looking at this, I've also noticed that we need to ensure that kata-deploy always points to the latest release, always, and that's something we don't currently do as on main it's pointing to the `rc0` release, even if `2.1.0` is out.

@jcvenegas, do you want me to break 3 out of this and open a different issue for that?

**Comment by c3d** [comment]
@fidencio Good idea, but wouldn't be useful to be able pass the next version to `tag_repos.sh`?

Also, do you need to take the minor version into account for stable branches? I assume you also run `tag_repos` for a minor version, right?

**Comment by fidencio** [comment]
> @fidencio Good idea, but wouldn't be useful to be able pass the next version to `tag_repos.sh`?

Why?
The process, as document [here|https://github.com/kata-containers/kata-containers/blob/main/docs/Release-Process.md] is:
* Run `update-respository-version.sh`, which will bump the current version to the next one.
* Run ./tag_repos.sh`, which will get the bumped version and create the tags based on that.

I'm not saying we shouldn't, I'm just trying to understand what's the benefit of adding this to the one doing the release and, maybe, end up with a wrong version (as happened in the past when doing this manually).

> Also, do you need to take the minor version into account for stable branches?

For stable branches it's less of a problem, but it's still a problem.
Maybe we should consider re-ordering what we do, as in:
* Currently:
  * We develop;
  * Bump the branch to the release version we want to release;
  * Release;

* Proposed:
  * After a full cycle of what we currently have ...
  * Bump the branch as soon as the release is cut;
  * Release;

What do you think?  The latter is what I usually do on the other projects I maintain.

This would help to keep the versioning right, but we'd have to tweak things on the kata-deploy side, as that has to point to the latest release (not to the one in development, till one commit before we do the release ;-)).

> I assume you also run `tag_repos` for a minor version, right?

We do,  and we also have the problem there.  However, it's less of a problem when we have 2.1.0+something, for instance.  That's understandable that it was something atop of 2.1.0 stable release.
When we have 2.1.0-rc0+something, that becomes hard to understand whether 2.1.0 material was included as part of that or not.
